slapo/schedule.py

```diff
@@ -525,103 +537,155 @@ def find_node(self, pattern):
+                break
+        else:
+            raise RuntimeError("Cannot find the first non-placeholder operator")
 
         res = []
         for parent_name, submod in self.mod.named_modules():
             if not isinstance(submod, fx.GraphModule):
                 continue
-
             for node in submod.graph.nodes:
-                name = (
-                    f"{parent_name}.{node.target}" if parent_name != "" else node.target
-                )
-                if (
-                    node.op == "placeholder"
-                    or not isinstance(name, str)
-                    or not re.match(mod_name_pat, name)
-                ):
+                if node.op == "placeholder":
                     continue
+                subgraph = []
+                target_node = first_op
+                curr_node = node
+                if find_match_subgraphs(curr_node, target_node, subgraph):
+                    res.append(subgraph.copy())
+        return res
 
-                if func_pattern is None:
-                    # only find module
-                    res.append((parent_name, node))
-                    continue
+    def find(self, regex_or_pattern_fn):
+        """Find a node or a subgraph in a static dataflow graph.
+        This API is a dispatcher for `find_node` and `find_subgraph`
 
-                subgraph = [(parent_name, node)]
-                for target_node in list(pattern_mod.graph.nodes):
-                    # get the first placeholder,
-                    # i.e., the input of the target graph
-                    if target_node.op == "placeholder":
-                        curr_node = node
-                        if find_match_subgraphs(curr_node, target_node, subgraph):
-                            res.append(subgraph)
-                        break
-                else:
-                    raise RuntimeError("Cannot find the first placeholder")
-        return res
+        If you need to match a general node pattern, please directly use the `find_node` API.
 
-    def find(self, node_pattern, func_pattern=None):
-        if isinstance(node_pattern, str):
-            return self.find_subgraph(node_pattern, func_pattern=func_pattern)
-        if isinstance(node_pattern, FunctionType):
-            return self.find_node(node_pattern)
-        raise RuntimeError(f"Unrecognized pattern {node_pattern}")
+        Parameters
+        ----------
+        regex_or_pattern_fn : Union[str, Callable]
+            A regular expression for specifying the target of `call_module` node, or
+            a callable function/Pattern class specifying the subgraph pattern
+        """
+        if isinstance(regex_or_pattern_fn, (FunctionType, Pattern)):
+            return self.find_subgraph(regex_or_pattern_fn)
+        if isinstance(regex_or_pattern_fn, str):
+            return self.find_node(regex_or_pattern_fn)
+        raise RuntimeError(f"Unrecognized pattern {regex_or_pattern_fn}")
```

Comment:
Better to show its type (`Unrecognized pattern type {type(regex_or_pattern_fn)}`) as you're using type for dispatching